lib/reverse_sentence.rb

```diff
@@ -1,6 +1,19 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2) in the worst case scenario and O(n) in the best case. But since we always consider the worst case, it's O(n^2).
+# Space complexity: O(1) No matter the sentence, one array gets made, that eventually gets converted back into a string.
+
 def reverse_sentence(my_sentence)
```

Comment:
This works, but has O(n) and space complexity.  Take a look at my comment below as to why.  There is a way to do this O(1) space complexity.  Consider:  Reverse all the letters in the sentence, then go through and reverse each individual word.   Can you see a way to do this?